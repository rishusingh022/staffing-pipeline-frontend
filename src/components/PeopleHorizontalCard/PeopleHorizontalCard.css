.user-horizontal-card-container{
  width: 100%;
  display: flex;
  background-color: white;
  min-width: 250px;
  padding: 10px;
  position: relative;
  box-shadow: 0px 3px 7px 0px rgba(0,0,0,0.2);
  align-items: center;
}

.user-horizontal-img-container{
  height: 70px;
  aspect-ratio: 1/1;
  overflow: hidden;
  border-radius: 50%;
  flex-shrink: 0;
  box-sizing: border-box;
}

.user-horizontal-img{
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.user-horizontal-details{
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-wrap: wrap;
  width: 100%;
}

.user-horizontal-id{
  font-size: 13px;
  font-weight: 600;
  font-weight: 400;
}
.user-horizontal-name{
  font-size: 20px;
  font-weight: 600;
}

.user-horizontal-position{
  font-size: 15px;
  font-weight: 600;
}

.user-horizontal-office{
  font-size: 12px;
  font-weight: 600;
  color: #8C8C8C;
}

.user-horizontal-bottom-btn{
  display: flex;
  justify-content: flex-end;
  position: absolute;
  bottom: 0;
  right: 0;
  padding: 10px;
}

.user-horizontal-bottom-btn > div{
  display: flex;
  align-items: center;
  gap: 5px;
  cursor: pointer;
}
.user-horizontal-bottom-btn  p{
  font-size: 12px;
  font-weight: 600;
}

.active{
  background-color: rgb(136, 206, 136);
}