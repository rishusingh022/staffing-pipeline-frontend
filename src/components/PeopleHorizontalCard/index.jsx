import React from 'react';
import './PeopleHorizontalCard.css';
import ArrowIcon from '../../assets/images/Vector.png';
import UserImage from '../../assets/images/user-default.png';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

function PeopleHorizontalCard({ userImage, userFMNO, userName, userPosition, userOffice, userId }) {
  const navigate = useNavigate();
  const handleKnowMore = () => {
    navigate(`/users/${userId}`);
  };
  return (
    <div className="user-horizontal-card-container">
      <div className="user-horizontal-img-container">
        <img className="user-horizontal-img" src={userImage ? userImage : UserImage}></img>
      </div>
      <div className="user-horizontal-details">
        <p className="user-horizontal-id">FMNO: {userFMNO}</p>
        <p className="user-horizontal-name">{userName}</p>
        <p className="user-horizontal-position">{userPosition}</p>
        <p className="user-horizontal-office">{userOffice}</p>
        <div className="user-horizontal-bottom-btn" onClick={handleKnowMore}>
          <div>
            <p>Know More</p>
            <img src={ArrowIcon}></img>
          </div>
        </div>
      </div>
    </div>
  );
}

PeopleHorizontalCard.propTypes = {
  userImage: PropTypes.string,
  userFMNO: PropTypes.string,
  userName: PropTypes.string,
  userPosition: PropTypes.string,
  userOffice: PropTypes.string,
  userId: PropTypes.string,
};

export default PeopleHorizontalCard;
