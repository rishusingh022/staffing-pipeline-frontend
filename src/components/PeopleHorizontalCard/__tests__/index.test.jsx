import React from 'react';
import PeopleHorizontalCard from '..';
import { render } from '@testing-library/react';

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

describe('PeopleHorizontalCard Snapshot', () => {
  it('should render', () => {
    const { asFragment } = render(<PeopleHorizontalCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
