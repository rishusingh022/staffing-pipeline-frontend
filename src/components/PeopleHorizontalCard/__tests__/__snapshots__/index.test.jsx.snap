// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleHorizontalCard Snapshot should render 1`] = `
<DocumentFragment>
  <div
    class="user-horizontal-card-container"
  >
    <div
      class="user-horizontal-img-container"
    >
      <img
        class="user-horizontal-img"
        src="user-default.png"
      />
    </div>
    <div
      class="user-horizontal-details"
    >
      <p
        class="user-horizontal-id"
      >
        FMNO: 
      </p>
      <p
        class="user-horizontal-name"
      />
      <p
        class="user-horizontal-position"
      />
      <p
        class="user-horizontal-office"
      />
      <div
        class="user-horizontal-bottom-btn"
      >
        <div>
          <p>
            Know More
          </p>
          <img
            src="Vector.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
